hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/s3guard/ITestDynamoDBMetadataStoreAuthoritativeMode.java

```diff
@@ -0,0 +1,558 @@
+        S3GUARD_DDB_BACKGROUND_SLEEP_MSEC_KEY,
+        METADATASTORE_AUTHORITATIVE,
+        AUTHORITATIVE_PATH);
+    conf.setTimeDuration(
+        S3GUARD_DDB_BACKGROUND_SLEEP_MSEC_KEY,
+        0,
+        TimeUnit.MILLISECONDS);
+    return conf;
+  }
+
+  @Override
+  public void setup() throws Exception {
+    super.setup();
+    S3AFileSystem fs = getFileSystem();
+    Configuration conf = fs.getConf();
+    S3ATestUtils.assumeS3GuardState(true, conf);
+    storeContext = fs.createStoreContext();
+    assume("Filesystem isn't running DDB",
+        storeContext.getMetadataStore() instanceof DynamoDBMetadataStore);
+    metastore = (DynamoDBMetadataStore) storeContext.getMetadataStore();
+    URI fsURI = storeContext.getFsURI();
+    fsUriStr = fsURI.toString();
+    if (!fsUriStr.endsWith("/")) {
+      fsUriStr = fsUriStr + "/";
+    }
+    auditor = new AuthoritativeAuditOperation(storeContext,
+        metastore, true);
+
+
+    if (authFS == null) {
+      // creating the test stores
+      basePath = path("base");
+      authPath = new Path(basePath, "auth");
+      nonauthPath = new Path(basePath, "nonauth");
+      final Configuration authconf = new Configuration(conf);
+      final URI uri = authPath.toUri();
+      authconf.set(AUTHORITATIVE_PATH, uri.toString());
+      authconf.setBoolean(METADATASTORE_AUTHORITATIVE, true);
+      authFS = (S3AFileSystem) FileSystem.newInstance(uri, authconf);
+
+      // and create the unguarded at the same time
+      final Configuration unguardedConf = new Configuration(conf);
+      removeBaseAndBucketOverrides(unguardedConf,
+          S3_METADATA_STORE_IMPL);
+      unguardedFS = (S3AFileSystem) FileSystem.newInstance(uri, unguardedConf);
+    }
+    cleanupMethodPaths();
+    dir = new Path(methodAuthPath, "dir");
+    dirFile = new Path(dir, "file");
+  }
+
+  @Override
+  public void teardown() throws Exception {
+    try {
+      cleanupMethodPaths();
+    } catch (IOException ignored) {
+    }
+    super.teardown();
+  }
+
+  /**
+   * Clean up from other test runs which halted.
+   * Uses the authfs; no-op if null.
+   * @throws IOException Failure
+   */
+  private void cleanupMethodPaths() throws IOException {
+    S3AFileSystem fs = authFS;
+    if (fs != null) {
+      methodAuthPath = new Path(authPath, getMethodName());
+      fs.delete(methodAuthPath, true);
+      methodNonauthPath = new Path(nonauthPath, getMethodName());
+      fs.delete(methodNonauthPath, true);
+    }
+  }
+
+  @Test
+  public void testEmptyDirMarkerIsAuth() {
+    final S3AFileStatus st = new S3AFileStatus(true, dir, "root");
+    final DDBPathMetadata md = (DDBPathMetadata) emptyDirectoryMarker(st);
+    Assertions.assertThat(md)
+        .describedAs("Metadata %s", md)
+        .matches(DDBPathMetadata::isAuthoritativeDir, "is auth dir")
+        .matches(d -> d.isEmptyDirectory() == Tristate.TRUE,
+            "isEmptyDirectory");
+  }
+
+  @Test
+  @Ignore("Needs mkdir to be authoritative")
```

Comment:
Do you want to ignore these tests? Is there a plan to make mkdir authoritative later?

Reply:
yes, filed JIRA and referenced., I will not fix here as the PR is big enough already